File: src/app/src/main/java/jp/co/e2/dogage/activity/SettingActivity.java
package jp.co.e2.dogage.activity;

import android.app.Activity;
import android.app.Fragment;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.design.widget.Snackbar;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.RelativeLayout;
import android.widget.TextView;

import jp.co.e2.dogage.R;
import jp.co.e2.dogage.common.AndroidUtils;
import jp.co.e2.dogage.common.PreferenceUtils;
import jp.co.e2.dogage.config.Config;

/**
 * 設定アクテビティ
 */
public class SettingActivity extends BaseActivity {
    /**
     * ファクトリーメソッドもどき
     *
     * @param activity アクテビティ
     * @return intent
     */
    public static Intent newInstance(Activity activity) {
        return new Intent(activity, SettingActivity.class);
    }

    /**
     * ${inheritDoc}
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_common);

        //アクションバーをセットする
        setBackArrowToolbar();

        if (savedInstanceState == null) {
            getFragmentManager().beginTransaction().add(R.id.container, SettingFragment.newInstance()).commit();
        }
    }

    /**
     * ${inheritDoc}
     */
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
        }

        return super.onOptionsItemSelected(item);
    }

    /**
     * SettingFragment
     */
    public static class SettingFragment extends Fragment {
        private View mView = null;

        /**
         * ファクトリーメソッド
         *
         * @return SettingFragment
         */
        public static SettingFragment newInstance() {
            Bundle args = new Bundle();

            SettingFragment fragment = new SettingFragment();
            fragment.setArguments(args);

            return fragment;
        }

        /**
         * ${inheritDoc}
         */
        @Override
        public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);

            mView = inflater.inflate(R.layout.fragment_setting, container, false);

            //コンテンツセット
            setContent();

            //イベントセット
            setClickEvent();

            return mView;
        }

        /**
         * コンテンツをセットする
         */
        private void setContent() {
            //誕生日通知チェックボックス
            CheckBox checkBoxBirthNotify = mView.findViewById(R.id.checkBoxBirthNotification);
            checkBoxBirthNotify.setChecked(PreferenceUtils.get(getActivity(), Config.PREF_BIRTH_NOTIFY_FLG, true));

            //命日通知チェックボックス
            CheckBox checkBoxArchiveNotify = mView.findViewById(R.id.checkBoxArchiveNotification);
            checkBoxArchiveNotify.setChecked(PreferenceUtils.get(getActivity(), Config.PREF_ARCHIVE_NOTIFY_FLG, true));

            //アプリバージョン
            TextView textViewVer = mView.findViewById(R.id.textViewVer);
            textViewVer.setText(AndroidUtils.getVerName(getActivity()));
        }

        /**
         * クリックイベントをセットする
         */
        private void setClickEvent() {
            //誕生日通知チェックボックス
            CheckBox checkBoxBirthNotify = mView.findViewById(R.id.checkBoxBirthNotification);
            checkBoxBirthNotify.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
                @Override
                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                    changeNotifySetting(Config.PREF_BIRTH_NOTIFY_FLG, isChecked);
                }
            });

            //誕生日通知チェックボックス
            CheckBox checkBoxArchiveNotify = mView.findViewById(R.id.checkBoxArchiveNotification);
            checkBoxArchiveNotify.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
                @Override
                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                    changeNotifySetting(Config.PREF_ARCHIVE_NOTIFY_FLG, isChecked);
                }
            });

            //年齢計算について
            RelativeLayout relativeLayoutCalcAge = mView.findViewById(R.id.relativeLayoutCalcAge);
            relativeLayoutCalcAge.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    startActivity(AboutActivity.newInstance(getActivity()));
                }
            });

            //E2リンク
            Button buttonProducedBy = mView.findViewById(R.id.buttonProducedBy);
            buttonProducedBy.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Uri uri = Uri.parse(Config.OFFICIAL_LINK);
                    Intent i = new Intent(Intent.ACTION_VIEW,uri);
                    startActivity(i);
                }
            });
        }

        /**
         * チェックボックスの設定を保存する
         *
         * @param name プリファレンス保存名
         * @param value プリファレンス保存値
         */
        private void changeNotifySetting(String name, boolean value) {
            PreferenceUtils.save(getActivity(), name, value);

            AndroidUtils.showSuccessSnackBarS(mView, getString(R.string.change_settings));
        }
    }
}

File: src/app/src/main/java/jp/co/e2/dogage/activity/SettingActivity.kt
package jp.co.e2.dogage.activity

import android.app.Fragment
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.support.constraint.ConstraintLayout
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.CheckBox
import android.widget.TextView

import jp.co.e2.dogage.R
import jp.co.e2.dogage.common.AndroidUtils
import jp.co.e2.dogage.common.PreferenceUtils
import jp.co.e2.dogage.config.Config

/**
 * 設定アクテビティ
 */
class SettingActivity : BaseActivity() {
    /**
     * ${inheritDoc}
     */
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_common)

        //アクションバーをセットする
        setBackArrowToolbar()

        if (savedInstanceState == null) {
            fragmentManager.beginTransaction().add(R.id.container, SettingFragment()).commit()
        }
    }

    /**
     * ${inheritDoc}
     */
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
        }

        return super.onOptionsItemSelected(item)
    }

    /**
     * SettingFragment
     */
    class SettingFragment : Fragment() {
        /**
         * ${inheritDoc}
         */
        override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
            super.onCreate(savedInstanceState)

            val view = inflater.inflate(R.layout.fragment_setting, container, false)

            //コンテンツセット
            setContent(view)

            return view
        }

        /**
         * コンテンツをセットする
         */
        private fun setContent(view: View) {
            //誕生日通知チェックボックス
            view.findViewById<CheckBox>(R.id.checkBoxBirthNotification).apply {
                this.isChecked = PreferenceUtils.get(activity, Config.PREF_BIRTH_NOTIFY_FLG, true)

                this.setOnCheckedChangeListener { buttonView, isChecked ->
                    changeNotifySetting(Config.PREF_BIRTH_NOTIFY_FLG, isChecked) }
            }

            //命日通知チェックボックス
            view.findViewById<CheckBox>(R.id.checkBoxArchiveNotification).apply {
                this.isChecked = PreferenceUtils.get(activity, Config.PREF_ARCHIVE_NOTIFY_FLG, true)

                this.setOnCheckedChangeListener { buttonView, isChecked ->
                    changeNotifySetting(Config.PREF_ARCHIVE_NOTIFY_FLG, isChecked) }
            }

            //アプリバージョン
            view.findViewById<TextView>(R.id.textViewVer).apply {
                this.text = AndroidUtils.getVerName(activity)
            }

            //年齢計算について
            view.findViewById<ConstraintLayout>(R.id.constraintLayoutCalcAge).apply {
                this.setOnClickListener {
                    startActivity(Intent(activity, AboutActivity::class.java))
                }
            }

            //E2リンク
            view.findViewById<Button>(R.id.buttonProducedBy).apply {
                this.setOnClickListener {
                    val i = Intent(Intent.ACTION_VIEW, Uri.parse(Config.OFFICIAL_LINK))
                    startActivity(i)
                }
            }
        }

        /**
         * チェックボックスの設定を保存する
         *
         * @param name プリファレンス保存名
         * @param value プリファレンス保存値
         */
        private fun changeNotifySetting(name: String, value: Boolean) {
            PreferenceUtils.save(activity, name, value)

            AndroidUtils.showSuccessSnackBarS(view, getString(R.string.change_settings))
        }
    }
}
